src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs

```diff
@@ -59,8 +59,47 @@ private void Start()
         {
             _logger.LogInformation($"Starting {nameof(LinuxContainerActivityPublisher)}");
 
-            // start the timer by setting the due time
-            SetTimerInterval(_initialFlushIntervalMs);
+            _publishingTask = StartPublishingAsync();
+        }
+
+        private async Task StartPublishingAsync()
+        {
+            _publishingCts = new CancellationTokenSource();
+
+            try
+            {
+                while (!_publishingCts.IsCancellationRequested)
+                {
+                    try
+                    {
+                        var nextFlushDelay = _flushIntervalMs;
+
+                        if (_initialPublish)
+                        {
+                            _initialPublish = false;
+                            await PublishSpecializationCompleteEvent();
+
+                            nextFlushDelay -= _initialFlushIntervalMs;
+                        }
+                        else
+                        {
+                            await FlushFunctionExecutionActivities();
+                        }
+
+                        await Task.Delay(nextFlushDelay, _publishingCts.Token);
+                    }
+                    catch (Exception ex)
```

Comment:
This catch is the actual fix, right? All the rest is just a refactoring away from a Timer based solution?

Looks like we're using async void timer callbacks in at least 2 other places (GrpcWorkerChannel and WorkerConcurrencyManager) and it looks like the the problem exists in the former as well (uncaught exceptions).

Reply:
I'm already following up on the other instances.